android/src/main/java/com/clevertap/clevertap_plugin/isolate/IsolateHandlePreferences.java

```diff
@@ -0,0 +1,47 @@
+package com.clevertap.clevertap_plugin.isolate;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import io.flutter.view.FlutterCallbackInformation;
+
+//TODO: make it singleton
```

Comment:
@shivamsharma2710 Resolve the TODO or remove it

Reply:
Resolved.